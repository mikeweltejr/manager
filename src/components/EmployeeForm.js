import React, { Component } from 'react';
import { View, Text, Picker, Platform } from 'react-native';
import { connect } from 'react-redux';
import { employeeUpdate } from '../actions';
import { CardSection, Input } from './common';

class EmployeeForm extends Component {
    render() {
        const { pickerContainerStyle, pickerTextStyle, pickerStyle  } = styles;

        return (
            <View>
                <CardSection>
                        <Input
                            label="Name"
                            placeholder="Joe Smith"
                            value={this.props.name}
                            onChangeText={text => this.props.employeeUpdate({ prop: 'name', value: text })}                     
                        />
                    </CardSection>
                    <CardSection>
                        <Input
                            label="Phone"
                            placeholder="555-555-5555"
                            value={this.props.phone}
                            onChangeText={value => this.props.employeeUpdate({ prop: 'phone', value })}
                        />
                    </CardSection>
                    <CardSection>
                        <View style={pickerContainerStyle}>
                            <Text style={pickerTextStyle}>Shift</Text>
                            <Picker
                                style={pickerStyle}
                                selectedValue={this.props.shift}
                                onValueChange={day => this.props.employeeUpdate({ prop: 'shift', value: day })}
                            >
                                <Picker.Item label="Monday" value="Monday" />
                                <Picker.Item label="Tuesday" value="Tuesday" />
                                <Picker.Item label="Wednesday" value="Wednesday" />
                                <Picker.Item label="Thursday" value="Thursday" />
                                <Picker.Item label="Friday" value="Friday" />
                                <Picker.Item label="Saturday" value="Saturday" />
                                <Picker.Item label="Sunday" value="Sunday" />                 
                            </Picker>
                        </View>
                    </CardSection>
                </View>
        );
    }
}

const styles = {
  pickerTextStyle: {
    fontSize: 18,
    paddingLeft: 20,
    ...Platform.select({
      android: {
        flex: 1
      }
    })
  },
  pickerStyle: {
    ...Platform.select({
      ios: {
        flex: 1
      },
      android: {
        color: '#000',
        paddingRight: 5,
        paddingLeft: 5,
        flex: 2
      }
    })
  },
  pickerContainerStyle: {
    ...Platform.select({
      ios: {
        flexDirection: 'column'
      },
      android: {
        height: 40,
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center'
      }
    })
  }
};

const mapStateToProps = state => {
    const { name, phone, shift } = state.employeeForm;
    return { name, phone, shift };
};

export default connect(mapStateToProps, { employeeUpdate })(EmployeeForm);