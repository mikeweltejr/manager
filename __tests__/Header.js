import 'react-native';
import React from 'react';
import { Header } from '../src/components/common';
import renderer from 'react-test-renderer';

test('Header renders correctly', () => {
    const tree = renderer.create(
        <Header />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
