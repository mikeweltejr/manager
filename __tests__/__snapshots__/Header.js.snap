// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F8F8F8",
      "elevation": 2,
      "height": 60,
      "justifyContent": "center",
      "paddingTop": 15,
      "position": "relative",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
      }
    }
  />
</View>
`;
